pkg/draft/pack/archive.go

```diff
@@ -0,0 +1,171 @@
+package pack
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+
+	"k8s.io/helm/pkg/proto/hapi/chart"
+)
+
+// Archive creates an archived pack to the given directory.
+//
+// This takes an existing pack and a destination directory.
+//
+// If the directory is /foo, and the pack is named bar, this
+// will generate /foo/bar.tgz.
+//
+// This returns the absolute path to the pack archive file.
+func Archive(p *Pack, outDir string) (string, error) {
+
+	// Create archive
+	if fi, err := os.Stat(outDir); err != nil {
+		return "", err
+	} else if !fi.IsDir() {
+		return "", fmt.Errorf("location %s is not a directory", outDir)
+	}
+
+	name := p.Metadata.Name
+	if name == "" {
+		return "", errors.New("no pack name specified")
+	}
+
+	filename := fmt.Sprintf("%s.tgz", name)
+	filename = filepath.Join(outDir, filename)
+	f, err := os.Create(filename)
+	if err != nil {
+		return "", err
+	}
+
+	// Wrap in gzip writer
+	zipper := gzip.NewWriter(f)
+	// zipper.Head.Extra = headerBytes // Helm uses this. Why?
+	zipper.Header.Comment = "Draft"
+
+	// Wrap in tar writer
+	twriter := tar.NewWriter(zipper)
+	rollback := false
+	defer func() {
+		twriter.Close()
+		zipper.Close()
+		f.Close()
+		if rollback {
+			os.Remove(filename)
+		}
+	}()
+
+	if err := writePackTarContent(twriter, p); err != nil {
+		rollback = true
+	}
+
+	return filename, err
+}
+
+func writePackTarContent(out *tar.Writer, p *Pack) error {
+
+	base := p.Metadata.Name
+
+	// write the metadata
+	pData, err := yaml.Marshal(p.Metadata)
+	if err != nil {
+		return err
+	}
+
+	if err := writeToTar(out, path.Join(base, MetadataName), pData); err != nil {
+		return err
+	}
+
+	// write the content of the chart
+	chartName := p.Chart.Metadata.Name
+	// HACK(rodcloutier): writeTarContents uses the chart name as the dirname. Because we want to
+	// write it to chart/, we name the chart 'chart'
```

Comment:
this is a good workaround. I've actually considered re-naming the chart name to "chart" to fix https://github.com/Azure/draft/issues/219#issuecomment-313554646 as well so we can run `helm package` on an application's chart, but that's off-topic to this conversation. Just saying that this is fine. :)